flang/lib/Lower/OpenMP.cpp

```diff
@@ -90,6 +123,8 @@ static void createBodyOfOp(Op &op, Fortran::lower::AbstractConverter &converter,
   }
   // Reset the insertion point to the start of the first block.
   firOpBuilder.setInsertionPointToStart(&block);
+  if (clauses)
```

Comment:
There can be other clauses in `OmpClauseList` as well ?
Ah, now I see you've removed some code while clauses parsing for filtering.
-- Is my understanding correct ?

Reply:
There can be others, but we will process clauses which are involved in privatisation (firstprivate, lastprivate, private etc). Others will be handled before creating the Operation.